velox/exec/LocalPlanner.cpp

```diff
@@ -381,6 +381,23 @@ void LocalPlanner::markMixedJoinBridges(
   }
 }
 
+namespace {
+
+// If the upstream is partial limit, downstream is final limit and we want to
+// flush as soon as we can to reach the limit and do as less work as possible.
+bool eagerFlush(const core::PlanNode& node) {
+  if (auto* limit = dynamic_cast<const core::LimitNode*>(&node)) {
+    return limit->isPartial() && limit->offset() + limit->count() < 10'000;
+  }
+  if (node.sources().empty()) {
+    return false;
+  }
+  // Follow the first source (e.g. probe side of join).
```

Comment:
Do we want to go below the join ? Any particular reason this function doesn't take core::PartitionedOutputNode as an argument and checks only it direct source node?

Reply:
There can be some nodes between partial limit and partitioned output.  For project it is obvious we should follow it, for join it is worth to follow the probe side (since it is the one trigger `getOutput` to spit out something).  I think it worth following as long as it is the one drives the generation of output.  Or do we want to limit it to certain node only, e.g. project/join?